### PR TITLE
Release 2.1

### LINKED ISSUE #196: cmake fails without boost
Is boost optional or no?

### PR DESCRIPTION
**Pull request by alkino**
Update release documentation according to commits.
We have a good deal of stuff:
* 4197dca (HEAD -> master, origin/master, origin/HEAD) Making optional Boost dependency non-default (fixes #196) (#199)
* 41a7ca1 Compile doxygen documentation with cmake (#220)
* 30ced28 Add dependency on hpc-conding-conventions (#236)
* d52344c Rewrite headers so that all public headers can be included (#225)
* 78a0d73 exist: always return true for / (#233)
* 5e32401 Manage HighFive dependencies in a clean way (#229)
* 9f05508 Cleaning up cmake with back-compat fixes (#230)
* 5291c0f Travis: add windows build
* b7a2e02 Getting type info and object kind (#221)
* 74250cf (origin/emmenlau_minor_cmake_improvements) Spelling correction comments (#210)
* 049c8c2 Fixing CI (#211)
* 6a4032e Avoid warning/error on downconvert for Intel (#205)
* 51d800c H5Easy: Support for n-d vector (#204)
* f49c02b Fix reading N dimension vector bug (#191)
* 5aecd86 Fixes in NodeTraits::getObjectName (#201)
* 7ab992b Fix issues with MSVC (#202)
* 49c96da Added missing inline to H5Easy (#198)
* e0667db Fix MSVC issue with 2D Vectors (#193)
* 036cfa1 Dropping invalid MSVC C++11 flag. Assume support (#192)
* 0a96afd CMakeLists.txt: Added c++ version support check for MSVC (#177)
* 7bb63fd Using latest OSX image (attempt) (#186)
* a9e0aaa Boost requirements depend on tests (#184)
* d135fec  workaround for Travis OSX build failures  (#182)
* 3086049 Fix another MSVC warnings when building for win32 (#179)
* 73d73ac Fix MSVC warnings (#169)
* e88b5b1 (bio/master) H5Easy: use standard ! operator (#175)
* 3383ced H5Easy API (#141)
* 68fbe8e (tklauser/master) Merge pull request #156 from tristan0x/more-compiler-flags
* ff62348 New compilation warnings / fix errors due to implicit conversions
* f000633 Merge pull request #162 from BioDataAnalysis/emmenlau_add_package_version
* e309b4c (bio/emmenlau_add_package_version) CMake/PackageConfig.cmake: added support for package version, and ensure PROJECT_VERSION is defined.
* fb79a7c Refactoring property lists (#157)
* 67a2ecf createGroup and exist can be recursive (#152)
* bf2db10 Merge pull request #153 from tristan0x/travis/several-compilers
* f2dbb6e travis: build with several compilers
* 92498f7 Replace static const members with unnamed enums
* 9fe0ec5 Merge pull request #148 from jwpeterson/patch-1
* f12d5bf Fix typo in README.md
* d2524b3 Merge pull request #145 from BlueBrain/cast_fix
* c3b757c Merge branch 'master' into cast_fix
* f259b82 Fix missing include for old versions of BOOST
* f87eca2 Do not use old style C cast
* 018d9bd explicitly override std::exception::what()
* a06a59c Fix for [1, N] -> [N] dimension conversion.
* a6867b9 Adding broken test
* 8c7a0c4 Adding Chunking, as well
* 0da510a Cleanup testing syntax
* 3b96240 Adding dataspace init with more than one parameter
* e472a7f Fixed implementation of user type construction for std::complex
* d971923 Shortcut syntax for dataset and attribute creation
* 3b4e9f3 Dropping old toggle for C++11
* 43bbb7f Adding test for missmatched diminsions for std::array
* 4921bd4 Adding support for std::array (#129)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
+    - Node_Traits gained getLinkType() and getObjectType()
+    - Node_Traits::createGroup() gained the parents argument to specify if intermediates groups should be created too
     - Added AnnotateTraits::createAttribute and NodeTraits::createDataset overloads to create and initialize the data in a single call
+    - Add the new H5Easy high level API.
+
+### FEATURE:
+    - Add a lot of new platform supported on travis (xcode, windows, clang, ...)
+    - DataSet now accept std::array and std::complex type
+    - Support for chunk cache configuration at data set create/open
+    - Add doc target to cmake and publish doc on: https://bluebrain.github.io/HighFive/
```
Comment (ferdonline) [inline CHANGELOG.md]: We should also tag these entries with the related issues.

Comment (ferdonline) [reply CHANGELOG.md]: Add a section Bug Fixes

**Comment by ferdonline** [comment]
Also @alkino please add yourself to the contributors list :)

File: CHANGELOG.md (hunk 1 above)
Comment (alkino) [reply CHANGELOG.md]: Can you create yourself the bug fixes section.
In fact, I tried, but it was not easy for me (I didn't know the library enough).

**Comment by ferdonline** [comment]
It's a pity the documentation ATM is only API docs. We should have examples, etc, specially for the new features... I guess we leave it for next release :/

**Comment by alkino** [comment]
For the documentation #222

**Comment by alkino** [comment]
LGTM, I merge it, push the tag and a new documentation once you review this.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,22 @@
...
-    - Support for std::array (unnested)
-    - Added AnnotateTraits::createAttribute and NodeTraits::createDataset overloads to create and initialize the data in a single call
+## Version 2.1 - 2019/10/XX
+### New Features:
+    - Inspection: API to get the type of links/objects and datasets data-types (#221)
+    - H5Easy: API for simple import/export to Eigen and xtensor (#141)
+    - Support for chunk and deflate configuration at dataset creation/open (#125). Added generic RawPropertyLists. (#157)
+    - Recursive `createGroup` and `exist` (#152)
+    - Shortcut syntax: ability to create a filled dataset in a single line (#130)
+    - DataSet now accepts std::complex and std::array's (#128, #129)
```
Comment (tristan0x) [inline CHANGELOG.md]:
```suggestion
    - DataSet now accepts `std::complex` and `std::array`'s (#128, #129)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,22 @@
...
+
+### Improvements
+    - Improved compat with MSVC and ICC compilers
+    - Cmake build system: modernized, create exported targets, better messages, etc.
```
Comment (tristan0x) [inline CHANGELOG.md]:
```suggestion
    - CMake build system: modernized, create exported targets, better messages, etc.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,7 +1,22 @@
...
+    - Building and publishing documentation: https://bluebrain.github.io/HighFive/
```
Comment (tristan0x) [inline CHANGELOG.md]: On https://bluebrain.github.io/HighFive : 

```
HighFive  3.0
HighFive - Header-only C++ HDF5 interface
```

3.0 ??

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,7 +1,22 @@
...
+    - Several other. See #231
+
+### Bug Fixes:
+    - Fixed header dependencies. They should all be include-able (#225)
```
Comment (tristan0x) [inline CHANGELOG.md]:
```suggestion
    - Fixed header dependencies. They are now all include-able (#225)
```

File: CHANGELOG.md (hunk 4 above)
Comment (alkino) [reply CHANGELOG.md]: It was a test to show the output. Once this patch is merged it will have the correct version number.

**Comment by alkino** [comment]
> I wish the Doxygen documentation provided the examples.

Let write it in #222

**Pull request merged**